### PR TITLE
Clarify 401 vs. 403

### PR DESCRIPTION
**Pull request by michielbdejong**
See https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error

### CODE REVIEW

**Comment by fkooman** [comment]
Straight from the horse's mouth (RFC 6750): 
```
   insufficient_scope
         The request requires higher privileges than provided by the
         access token.  The resource server SHOULD respond with the HTTP
         403 (Forbidden) status code and MAY include the "scope"
         attribute with the scope necessary to access the protected
         resource.
```

**Comment by michielbdejong** [comment]
Thanks @fkooman, fixed in this PR (needs squashing of the two commits).

**Comment by untitaker** [comment]
Same as #96?

On 6 November 2015 10:19:28 CET, Michiel de Jong <notifications@github.com> wrote:
>See
>https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error
>You can view, comment on, or merge this pull request online at:
>
>  https://github.com/remotestorage/spec/pull/105
>
>-- Commit Summary --
>
>  * Clarify 401 vs. 403
>
>-- File Changes --
>
>    M source.txt (5)
>
>-- Patch Links --
>
>https://github.com/remotestorage/spec/pull/105.patch
>https://github.com/remotestorage/spec/pull/105.diff
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/remotestorage/spec/pull/105

-- 
Sent from my phone. Please excuse my brevity.

File: source.txt [hunk 1]
```diff
@@ -281,11 +281,10 @@ Table of Contents
        * 414 if the request URI is too long,
        * 416 if Range requests are supported by the server and the Range
              request can not be satisfied,
-       * 401 for all requests that don't have a bearer token with
-             sufficient permissions,
-       * 403 for all requests that either have insufficient scope, e.g. 
-             accessing a <module> for which no scope was obtained, or accessing
-             data outside the user's <storage_root>,
+       * 401 for all requests that don't have a valid bearer token,
```
Comment (untitaker) [inline source.txt]: For which definition of "valid" though? Should perhaps be something like "for all requests with missing, expired or revoked token".

**Comment by michielbdejong** [comment]
Yes, but additionally if fixes https://github.com/remotestorage/spec/pull/96/files#r44145820 and removes the stray 'either' from the 403 text.

**Comment by michielbdejong** [comment]
Updated with phrasing from irc (because public documents don't require a valid bearer token), and with reference to [OAUTH].

File: source.txt [hunk 2]
```diff
@@ -281,11 +281,11 @@ Table of Contents
        * 414 if the request URI is too long,
        * 416 if Range requests are supported by the server and the Range
              request can not be satisfied,
-       * 401 for all requests that don't have a bearer token with
-             sufficient permissions,
-       * 403 for all requests that either have insufficient scope, e.g. 
-             accessing a <module> for which no scope was obtained, or accessing
-             data outside the user's <storage_root>,
+       * 401 for all requests that require a valid bearer token and
```
Comment (untitaker) [inline source.txt]: I still don't understand what "valid" is supposed to mean in this context. Syntactic validity might e.g. only mean that *some value* was sent. I couldn't find anything about valid tokens in the OAuth RFC.

Comment (michielbdejong) [inline source.txt]: It was [BEARER], not [OAUTH], sorry (it's also already mentioned at the start of the section).

**Comment by fkooman** [comment]
+1

**Pull request merged**